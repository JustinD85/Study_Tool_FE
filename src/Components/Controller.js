import React, { Component } from 'react';

class Controller extends Component {
  constructor(props) {
    super(props);
    this.state = {
      styles: {
        top: 50,
        left: 50
      }
    }

  }

  check = () => {
    const { intervalHasStarted } = this.state;

    if (!intervalHasStarted) {
     const stopInterval= setInterval(() => {
        console.log('every second every minute')
     }, 1000)
      this.setState({
        intervalHasStarted:true
      })
    }
  }

  render() {
    this.check();
    const { styles } = this.state;

    return (
      <span
        className="Controller"
        style={styles}
      > RAWR</span>
    )
  }
}

export default Controller;