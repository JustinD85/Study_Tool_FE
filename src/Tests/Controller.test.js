import React from 'react';
import Controller from '../Components/Controller';
import { shallow } from 'enzyme';

describe('<Controller/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Controller />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});