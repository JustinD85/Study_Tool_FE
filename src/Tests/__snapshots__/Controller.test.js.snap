// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controller/> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controller />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": " RAWR",
      "className": "Controller",
      "style": Object {
        "left": 50,
        "top": 50,
      },
    },
    "ref": null,
    "rendered": " RAWR",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": " RAWR",
        "className": "Controller",
        "style": Object {
          "left": 50,
          "top": 50,
        },
      },
      "ref": null,
      "rendered": " RAWR",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
